### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by apruneda**
Changes:
- Did an overall edit of the README and CONTRIBUTING pages
- Updated URLs
- Updated branding (Splunk Enterprise SDK)
- Fixed formatting issues
- Added a CREDITS.md page for third-party software credits

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,42 +2,34 @@
...
+If you're seeing some unexpected behavior with this project, please create an [issue](https://github.com/splunk/splunk-sdk-javascript/issues) on GitHub with the following information:
 
 1. Version of this project you're using (ex: 1.7.1)
-1. Platform version (ex: Windows Server 2012)
-1. Framework version (ex: Node.js 0.10.37) or Browser (ex: Chrome 43.0.2357.81)
-1. Splunk version (ex: 6.2.2)
-1. Other relevant information (ex: local/remote environment, Splunk network configuration)
+2. Platform version (ex: Windows Server 2012)
+3. Framework version (ex: Node.js 0.10.37) or Browser (ex: Chrome 43.0.2357.81)
+4. Splunk Enterprise version (ex: 8.0)
```
Comment (tdhellmann) [inline CONTRIBUTING.md:14]: Our CI runs against Splunk Enterprise 7.0 and 7.2

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,42 +2,34 @@
...
-If you're seeing some unexpected behavior with this project, please create an [issue on GitHub][issues] with the following information:
+If you're seeing some unexpected behavior with this project, please create an [issue](https://github.com/splunk/splunk-sdk-javascript/issues) on GitHub with the following information:
 
 1. Version of this project you're using (ex: 1.7.1)
-1. Platform version (ex: Windows Server 2012)
-1. Framework version (ex: Node.js 0.10.37) or Browser (ex: Chrome 43.0.2357.81)
-1. Splunk version (ex: 6.2.2)
-1. Other relevant information (ex: local/remote environment, Splunk network configuration)
+2. Platform version (ex: Windows Server 2012)
+3. Framework version (ex: Node.js 0.10.37) or Browser (ex: Chrome 43.0.2357.81)
```
Comment (tdhellmann) [inline CONTRIBUTING.md:13]: Tested Node versions are 0.12, 4.2, and 10.0 - we should probably list one of those

**Review by tdhellmann — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (apruneda) [reply CONTRIBUTING.md:14]: I changed the example to 7.0

File: CONTRIBUTING.md (hunk 2 above)
Comment (apruneda) [reply CONTRIBUTING.md:13]: got it

**Pull request merged**